lib/topic_query.rb

```diff
@@ -1284,10 +1284,6 @@ def filter_by_tags(result)
         # ANY of the given tags:
         result = result.joins(:tags).where("tags.id in (?)", tags)
       end
-
-      # TODO: this is very side-effecty and should be changed
-      # It is done cause further up we expect normalized tags
-      @options[:tags] = tags
```

Comment:
This was not a comment left by me but I did see this in the refactor I did. I think I tried removing it but it didn't seem to be so straightforward.